### DISCUSSION TITLE
How can I export the whole database to a file?

### DISCUSSION THREAD

**Discussion by yuanhoujun** [isar/isar#569] [Help]
Any API can do this?

**Comment by EArminjon** [comment]
Up :) ?

We plan to use this package and this feature is important for us. Any system to export and import from a file ?

**Reply by SupremeDeity** [reply]
There are two ways to do this.

- On the newer versions there is a `copyToFile` instance method on the Isar object. I'm not sure where the import option for this is but its a compacted version of the database. 

- The second method which I'm not sure if its the recommended method or not but I personally use is to basically use `importJson`/`exportJson` methods on each individual collection.

**Reply by abdullah-cse** [reply]
Will you share a sample code, where you used `importJson/exportJson`?

**Reply by derrickgw** [reply]
It is very similar to writing your own `toString` function.  I haven't needed to import JSON, but this is what some of my export functions look like:
```
@embedded
class Location {
  double lat = double.nan;
  double long = double.nan;

  Location();

  @override
  String toString() {
    return "Location: $lat,$long";
  }

  Map<String, dynamic> toJsonMap() {
    return {
      "latitude": lat.toString(),
      "longitude": long.toString(),
    };
  }
}
```
and
```
  Map<String, dynamic> toJsonMap() {
    return {
      "id": id,
      "sourceId": sourceId,
      "caption": caption,
      "timestamp": timestamp.toIso8601String(),
      "location": location.toJsonMap(),
      "features": features.map((e) => e.toJsonMap()).toList(),
    };
  }
```

**Reply by Ismaelito86** [reply]
Hi, I’m new in this ,I'm sorry if my question is a little bit silly. What’s the procedure to initialize the Isar database in the app with default values? Thanks a lot

**Comment by Ismaelito86** [comment]
Hi, I’m new in this ,I'm sorry if my question is a little bit silly. What’s the procedure to initialize the Isar database in the app with default values? Thanks a lot
